File: Backend/src/Sofco.Model/Models/Menu.cs
using Sofco.Model.Interfaces;
using System;
using System.Collections.Generic;

namespace Sofco.Model.Models
{
    public class Menu : BaseEntity, ILogicalDelete
    {
        public string Description { get; set; }

        public string Url { get; set; }

        public string Code { get; set; }

        public bool Active { get; set; }

        public IList<Module> Modules { get; set; }
    }
}

File: Frontend/src/models/menu.ts
import { Module } from 'models/module';
export class Menu{
    constructor(
        public id: number,
        public code: string = "",
        public description: string = "",
        public active: boolean = true,
        public url: string = '',
        public modules: Module[],
        public available: boolean = false
    ){}
    
}